src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/BrowserWebSockets/BrowserWebSocket.cs

```diff
@@ -0,0 +1,490 @@
+            if (_onError != null)
+            {
+                _innerWebSocket?.SetObjectProperty("onerror", "");
+                _onError = null;
+            }
+            if (_onOpen != null)
+            {
+                _innerWebSocket?.SetObjectProperty("onopen", "");
+                _onOpen = null;
+            }
+            if (_onMessage != null)
+            {
+                _innerWebSocket?.SetObjectProperty("onmessage", "");
+                _onMessage = null;
+            }
+        }
+
+        public override void Dispose()
+        {
+            int priorState = Interlocked.Exchange(ref _state, (int)InternalState.Disposed);
+            if (priorState == (int)InternalState.Disposed)
+            {
+                // No cleanup required.
+                return;
+            }
+
+            // registered by the CancellationTokenSource cts in the connect method
+            _cts.Cancel(false);
+            _cts.Dispose();
+
+            _writeBuffer?.Dispose();
+            _receiveMessageQueue.Writer.Complete();
+
+            NativeCleanup();
+
+            _innerWebSocket?.Dispose();
+        }
+
+        // This method is registered by the CancellationTokenSource cts in the connect method
+        // and called by Dispose or Abort so that any open websocket connection can be closed.
+        private async void AbortRequest()
+        {
+            if (State == WebSocketState.Open)
+            {
+                await CloseAsyncCore(WebSocketCloseStatus.NormalClosure, SR.net_WebSockets_Connection_Aborted, CancellationToken.None).ConfigureAwait(continueOnCapturedContext: true);
+            }
+        }
+
+        /// <summary>
+        /// Send data on <see cref="System.Net.WebSockets.ClientWebSocket"/> as an asynchronous operation.
+        /// </summary>
+        /// <returns>The async.</returns>
+        /// <param name="buffer">Buffer.</param>
+        /// <param name="messageType">Message type.</param>
+        /// <param name="endOfMessage">If set to <c>true</c> end of message.</param>
+        /// <param name="cancellationToken">Cancellation token.</param>
+        public override Task SendAsync(ArraySegment<byte> buffer, WebSocketMessageType messageType, bool endOfMessage, CancellationToken cancellationToken)
+        {
+            ThrowIfNotConnected();
+
+            if (messageType != WebSocketMessageType.Binary &&
+                    messageType != WebSocketMessageType.Text)
+            {
+                throw new ArgumentException(SR.Format(SR.net_WebSockets_Argument_InvalidMessageType,
+                    messageType,
+                    nameof(SendAsync),
+                    WebSocketMessageType.Binary,
+                    WebSocketMessageType.Text,
+                    nameof(CloseOutputAsync)),
+                    nameof(messageType));
+            }
+
+            WebSocketValidate.ValidateArraySegment(buffer, nameof(buffer));
+
+            _writeBuffer ??= new MemoryStream();
```

Comment:
Is this intentional? Caling SendAsync repeatably will grow MemoryStream with no limit

Reply:
The write buffer is nulled out below so it will be recreated.

Reply:
then why do you need `??=`

Reply:
because the message needs to be buffered until end of message is reached.  Once end of message is reached it gets nulled out and needs to be recreated.